### ISSUE TITLE
Question about DQMHarvest jobs

### ISSUE THREAD

**Issue by andresfelquintero** [dmwm/WMCore#8839]
Can DQMHarvest jobs fail in the upload to the GUI, but still be considered successful?

Because Data certification team has found that for  2 collision runs 320824 and 321781, some DQM files were not successfully transferred from T0 to DQM GUI. Basically these PDs

1) JetHT PD  for 321781
2) ZeroBias PD for 320824

It was found by Marcel that these jobs ended during the CMSWEB update downtime, one on 2018-08-07 and one on 2018-09-04,  and DQMGUI was probably down during that period which lead to this unsuccessful transfer.

**Comment by amaltaro** [comment]
Hi Andres, we haven't seen such cases in production and I believe it cannot happen.
However, looking at the source code
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/WMSpec/Steps/Executors/DQMUpload.py#L152

it seems we do not validate the http exit code though (we only fail uploads when an exception is raised). The best would be to have access to those jobs/logs.

Would you know the exact T0 workflow name for those harvesting jobs? Such that we can look for them in the EOS logs area.

**Comment by andresfelquintero** [comment]
Hello

The workflow names are JetHT  for run 321781 and ZeroBias for run 320824

**Comment by andresfelquintero** [comment]
Hello Alan. The workflows are /JetHT/Run2018D-PromptReco-v2/DQMIO for run 321781 and /JetHT/Run2018D-PromptReco-v2/DQMIO for run 320824. I don't know if this is the information that you requested in the proper format. 
Thanks.

**Comment by amaltaro** [comment]
I searched these runs in the tier0_wmstats, but I guess those runs/workflows have been already archived.
Chatting with Seangchan, we believe there would be an exception if the http request failed (but I'm still not 100% sure). I'll check with the T0 Ops to see if they can find the job logs.

**Comment by andresfelquintero** [comment]
Hello again Alan. Those runs have been archived, I think they should look like this PromptReco_Run320824_ZeroBias and PromptReco_Run321781_JetHT . Let me check if I can get the agent that was handling those workflows.

**Comment by andresfelquintero** [comment]
I have just confirmed that the agent that was handling this workflow was shut down and it does not exist anymore.

**Comment by amaltaro** [comment]
Haaa, look what I found in the EOS log area:
https://eoscmshttp.cern.ch/eos/cms/store/logs/prod/recent/PromptReco/PromptReco_Run320824_ZeroBias/RecoMergewrite_DQMIOEndOfRunDQMHarvestMerged

indeed the upload failed and we didn't catch that error, see the wmagentJob.log
```
2018-08-07 08:46:33,017:INFO:DQMUpload:HTTP upload finished succesfully with response:
Status code: None
Message: None
Detail: None
Data: <!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.1//EN" "http://www.w3.org/TR/xhtml11/DTD/xhtml11.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en">
 <head>
 <title>CMSWEB Error: Service unavailable</title>
...
```

I'll get it fixed on Monday.

**Comment by amaltaro** [comment]
@andresfelquintero @hufnagel you might want to patch the T0 agent to avoid such issues. Fix is in #8844
